target/linux/ath79/generic/base-files/etc/board.d/01_leds

```diff
@@ -225,6 +225,12 @@ glinet,gl-x750)
 hak5,lan-turtle)
 	ucidef_set_led_netdev "wan" "WAN" "orange:system" "eth1"
 	;;
+joyit,jt-or750i)
+	ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth0"
```

Comment:
@adschm The new bootloader sets GPIO MUX functions for the LAN/WAN ports' LEDs to blink. From what I heard it was preferred to reset the GPIO MUX registers and let kmod-ledtrigger do the work. Actually I've thought this was default in OpenWrt, but they are not. That's why I didn't realize the WAN-LED assignment was wrong. How shall this be handled? As the LEDs are integrated in the ports I don't expect anyone wanting to reassign them. Therefore I'd like to remove the GPIO network port LED definitions completely to save some CPU cycles...

Reply:
@blocktrron Do you have a clue how to handle this? From what I remember the GPIO MUX functions were resetted previously in ath79. Momentarily they are not. I'd prefer letting the MUX do the blinking. Is it okay to remove the port LED definitions or will OpenWrt eventually reset the GPIO MUX functions in the future again?

@s-2 You once showed me an ath79 board where they set the switch LED_BLINK and GPIO MUX registers from the DTS. Which was it? I think that would be an alternative approach so that when OpenWrt decides to reset the MUX register again, it won't break this board...

Reply:
> You once showed me an ath79 board where they set the switch LED_BLINK and GPIO MUX registers from the DTS.

Regarding the GPIO settings on the SoC side: the `&pinmux` node in `qca953x.dtsi` is set to `0x1804002c`, i.e. the start offset of `GPIO_OUT_FUNCTION0`; e.g. in compex_wpj531-16m: `<0x8 0x2b000000 0xff000000>` means GPIO 11 (in `GPIO_OUT_FUNCTION_2`) is set to `43: LED_LINK` (for port 2) etc.

Many other devices just write values to 0x0 to restore normal SoC GPIO function (overriding bootloader defaults, e.g.
`pinctrl-single,bits = <0xc 0x0 0x00ff0000>;`).

However, I can't seem to find devices changing LED registers of `builtin_switch` though, there are only initvals for external ar8327-switches (a few devices declare `pinmux` in the `builtin_switch` node, but that again refers to SoC regs eventually).

Technically, they should be accessible though, by writing `pinmux` entries for switch registers `0xB0, 0xB4, 0xB8, 0xBC`.
Not sure if those are the ones you mean? (they are 32 bit registers, accessible via tunneling of two 16 bit commands through MDIO), while there are also `LED_CONTROL` rules, which are 16bit registers on MDIO.
However, apparently none of these are ever written from any dts though...

Reply:
@s-2 Yes those are the registers I was referring to...

I'd prefer then to set the MUX registers in the DTS to reflect the bootloader's configuration and completely remove the GPIO LED definitions for the port LEDs. I think it is a waste of CPU cycles to trigger them and I can't figure out a reason someone wants to control them manually.